VideoLocker/src/main/java/org/edx/mobile/module/notification/NotificationPreference.java

```diff
@@ -0,0 +1,85 @@
+package org.edx.mobile.module.notification;
+
+import org.edx.mobile.model.api.CourseEntry;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * We extends HashMap to make the implementation extendible
+ */
+public class NotificationPreference extends ArrayList<EdxLocalParseChannel> implements Serializable{
```

Comment:
I can't find a really good article on this (here is a mediocre one http://blog.danlew.net/2010/05/23/android_serializable_considered_harmful/) , but I think we should avoid "Serializable" for anything that survives beyond the length of the process. It forces your persistent store to make a lot of assumptions about the underlying structure of a type - for example, you can no longer even rename member variables. We should probably use json or something like that.